src/App.vue

```diff
@@ -0,0 +1,343 @@
+			newTopUp[mountIdx] = x;
+			this.topUpValue = newTopUp;
+		},
+		async topupBatch(mountIdx, batchIdx, activeBatchId) {
+			const requestOptions = this.getRequestOptions(this.parsedMounts[mountIdx].mount_urloptions.user, this.parsedMounts[mountIdx].mount_urloptions.password);
+			const endpoint = this.parsedMounts[mountIdx].mount_urloptions.ip + ":" + this.parsedMounts[mountIdx].mount_urloptions.debug_api_port;
+			console.log("batch,amount=" + activeBatchId + "," + Number(this.topUpValue[mountIdx]));
+
+			try {
+				this.beeClient = new BeeDebug(endpoint);
+				await this.beeClient.topUpBatch(activeBatchId, Number(this.topUpValue[mountIdx]), requestOptions);
+			}
+			catch (err) {
+				console.log(err);
+			}
+		},
+		async buyPostage(mountidx, evt) {
+			if (evt) {
+				evt.preventDefault();
+			}
+			let newBatchlabel = [...this.newBatchLabel];
+			newBatchlabel[mountidx] = "Status...";
+			this.newBatchLabel = newBatchlabel;
+
+			this.newBatchBtnDisabled[mountidx] = true;
+			let endpoint = this.parsedMounts[mountidx].mount_urloptions.ip + ":" + this.parsedMounts[mountidx].mount_urloptions.debug_api_port;
+			if (!endpoint.toLowerCase().startsWith("http")) {
+				endpoint = "http://" + endpoint;
+			}
+			const requestOptions = this.getRequestOptions(this.parsedMounts[mountidx].mount_urloptions.user, this.parsedMounts[mountidx].mount_urloptions.password);
+			console.log("batch,depth=" + this.newBatchAmounts[mountidx] + "," + this.newBatchDepths[mountidx]);
+
+			newBatchlabel = [...this.newBatchLabel];
+			try {
+				this.beeClient = new BeeDebug(endpoint);
+
+				// this.newBatchLabel[mountidx] = await this.beeClient.getAllPostageBatch(requestOptions);
+				const newBatchId = await this.beeClient.createPostageBatch(Number(this.newBatchAmounts[mountidx]), Number(this.newBatchDepths[mountidx]), requestOptions);
+
+				newBatchlabel[mountidx] = "Success: Created new batch " + newBatchId;
+				this.newBatchLabel = newBatchlabel;
+
+				this.parsedMounts[mountidx].batches.push({ batchID: newBatchId, amount: this.newBatchAmounts[mountidx], batchTTL: '', isActive: false, isDisabled: false, });
+
+				this.newBatchBtnDisabled[mountidx] = false;
+			}
+			catch (err) {
+				console.log(err);
+				newBatchlabel[mountidx] = err;
+				this.newBatchLabel = newBatchlabel;
+				this.newBatchBtnDisabled[mountidx] = false;
+			}
+		},
+		async saveSettings(mountidx, evt) {
+			if (evt) {
+				evt.preventDefault();
+			}
+			this.setSaveBtnValue(mountidx, "Saving...");
+			this.setSaveMessage(mountidx, "");
+			this.saveSettingsBtn[mountidx] = true;
+
+			const url = generateUrl('/apps/files_external_ethswarm/save');
+			const parsedMountsToSave = this.parsedMounts.map(mount => ({ mount_id: mount.mount_id, encrypt: mount.encrypt, batchid: mount.batchid }));
+			console.log("json=" + JSON.stringify(this.parsedMounts) + ";len=" + this.parsedMounts.length + ";url=" + url + ";newparse=" + JSON.stringify(parsedMountsToSave));
```

Comment:
I would remove all console logs or wrap them in a debug mode flag.

Reply:
Implemented an override of the console function.
It allows the console.log lines to remain in place but the output depends on the true/false setting of the debug mode flag (debugConsole)

Reply:
Is this adequate or a bit of an overkill?